lib/ssl/test/ssl_cert_tests.erl

```diff
@@ -96,7 +98,42 @@ auth(Config) ->
                   end,
     ServerOpts =  [{verify, verify_peer} | ssl_test_lib:ssl_options(extra_server, server_cert_opts, Config)],
     ssl_test_lib:basic_test(ClientOpts, ServerOpts, Config).
+%%--------------------------------------------------------------------
+client_auth_custom_sign() ->
+    [{doc,"Test that client and server can connect using their own signature function"}].
+
+client_auth_custom_sign(Config) when is_list(Config) ->
+    KeyAlg =  proplists:get_value(cert_key_alg, Config),
+    % 'rsa_pss_rsae' is not an expected config value here, 
+    % using rsa or rsa_pss_pss instad seams to work,
+    % Not sure how to modify SSL to add it in the acceptable key pair groups...
+    CompatibleKeyAlg = case KeyAlg of rsa_pss_rsae -> rsa_pss_pss; _ -> KeyAlg end,
```

Comment:
I am in doubt on this, should I extend the group pairs, convert in the lib, or require the user to handle it as I do in this test?

Reply:
rsa_pss_pss and rsa_pss_rsae are only vaild for TLS-1.2 and TLS-1.3!

Reply:
Good to know, but this line only does a translation of the KeyAlg provided by the test group settings.
So those settings are right I presume. 
So this is not wrong, right? It could be moved to an internal translation as I suggest above.

Reply:
es we will need to specify that a key can be of type rsa_pss_pss. But rsa_pss_rsae is just a plain
rsa-key used with special options. Here there is a consideration that the person suppling the fun
must supply a fun that can handle rsa_pss_rsae options or they have to disable this signature_algorithm
if TLS-1.2 and/or TLS-1.3 are supported.

Reply:
I think this is something that any way should be added to the engine support (even without this PR)  and was sort of overlooked when introducing the rsa_pss_pss support.  I know both rsa_pss_pss and rsa_pss_rsae use the same option
list. That is related to the the crypto API. The big difference beeing that pss_pss  signature also uses the RSA-keys special params, that do not exist in the pss_rsae case.